R/get_plot_component.R

```diff
@@ -0,0 +1,42 @@
+#' Get plot components
+#'
+#' @param plot a ggplot or gtable
+#' @param pattern the name of the component
+#'
+#' @return a grob
+#' @export
+get_plot_component <- function(plot, pattern) {
+
+  grob_names <- plot_component_names(plot)
+  grobs <- plot_components(plot)
+
+  grobIndex <- which(grepl(pattern, grob_names))
+
+  if (length(grobIndex) != 0) {
+    if (length(grobIndex) > 1) {
+        warning("multiple components found; returning the first one")
```

Comment:
I think it would be better to return all components that are found, at least as an option. If you want, you can make this configurable by adding a parameter `return_all` or similar and set it to `FALSE` by default.